liquidity-rewards/contracts/UniswapPoolRewards.sol

```diff
@@ -1,5 +1,193 @@
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-contract UniswapPoolRewards {
-  
-}
+import "./Dependencies/LPTokenWrapper.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+
+contract UniswapPoolRewards is LPTokenWrapper, Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    string constant public NAME = "UniswapPoolRewards";
+
+    IERC20 public elimuToken;
+
+    uint256 public duration;
+    uint256 public periodFinish = 0;
+    uint256 public rewardPeriod = 0;
+
+    uint256 public rewardRate = 0;
```

Comment:
@hhio618 Can you add some documentation here explaining how the `rewardRate` works?

Reply:
The value of `rewardRate` is as follows:
`(total reward available on the contract)/ (reward duration)`
this will be used for calculating the `rewardPerTokenStored` variable, which is the rate of reward token per lp token stored on the contract.

Reply:
This way we can calculate the exact lp users' rewards in any amount of time they staked their tokens.

Reply:
@hhio618 Thank you. Can you add this information to the code?

```solidity
/**
 * @dev The value of rewardRate is as follows: ...
 */
uint256 public rewardRate = 0;
```

Reply:
> The value of `rewardRate` is as follows:
> `(total reward available on the contract)/ (reward duration)`

@hhio618 Since we are moving away from the "reward period" concept, there will be no "reward duration", correct? Instead, and if you think it makes sense, we can set a `rewardRate` _per Ethereum block_. Right now the average [time per block](https://etherscan.io/chart/blocktime) is ~15 seconds, which is:

- 4 blocks per minute
- 240 blocks per hour
- 5,760 blocks per day
- 172,800 blocks per month

So we can use this to calculate a reward rate of [322,500](https://voice.aragon.org/processes/#/bd8e3ed1c4ba7ce98067617813a6d2cd32cf951bf66f8bc3132724c5a04a2b61) `$ELIMU` / 172,800 blocks/month = **1.87** `$ELIMU`/block

Reply:
Ok, it seems good. but do we need to change this variable later, maybe whenever we wanted to increase the rewards?

Reply:
@hhio618 Yes, contract owners need to be able to adjust the reward rate.